### PR TITLE
feat: default sync provider configuration

### PR DESCRIPTION
**Pull request by james-milligan**
## This PR

- adds configuration for default sync provider 
- extends helm values to provide access to this configuration

### Related Issues

https://github.com/open-feature/open-feature-operator/issues/318

### CODE REVIEW

File: apis/core/v1alpha1/flagsourceconfiguration_types.go [hunk 1]
```diff
@@ -157,6 +173,10 @@ func (fc *FlagSourceConfigurationSpec) Merge(new *FlagSourceConfigurationSpec) {
 	if new.SyncProviderArgs != nil && len(new.SyncProviderArgs) > 0 {
 		fc.SyncProviderArgs = append(fc.SyncProviderArgs, new.SyncProviderArgs...)
 	}
+	fmt.Println("this is what is wrong", new.DefaultSyncProvider)
```
Comment (beeme1mr) [inline apis/core/v1alpha1/flagsourceconfiguration_types.go:176]: Please review this line or update the text.

File: chart/open-feature-operator/README.md [hunk 2]
```diff
@@ -49,9 +49,10 @@ The command removes all the Kubernetes components associated with the chart and
...
 | `sidecarConfiguration.socketPath`      | `""`  | Sets the value of the `XXX_SOCKET_PATH` environment variable for the injected sidecar container.|
-| `sidecarConfiguration.repository`      | `ghcr.io/open-feature/flagd`  | Sets the image for the injected sidecar container. |
-| `sidecarConfiguration.tag`      | current flagd version: `v0.3.2`  | Sets the version tag for the injected sidecar container. |
+| `sidecarConfiguration.image.repository`      | `ghcr.io/open-feature/flagd`  | Sets the image for the injected sidecar container. |
+| `sidecarConfiguration.image.tag`      | current flagd version: `v0.3.2`  | Sets the version tag for the injected sidecar container. |
 | `sidecarConfiguration.providerArgs`      | `""`  | Used to append arguments to the sidecar startup command. This value is a comma separated string of key values separated by '=', e.g. `key=value,key2=value2` results in the appending of `--sync-provider-args key=value --sync-provider-args key2=value2` |
+| `sidecarConfiguration.defaultSyncProvider`      | `kubernetes`  | Sets the value of the `XXX_SYNC_PROVIDER` environment variable for the injected sidecar container. |
```
Comment (beeme1mr) [inline chart/open-feature-operator/README.md:55]: Where can someone find a list of valid sync providers.

File: apis/core/v1alpha1/flagsourceconfiguration_types.go [hunk 3]
```diff
@@ -26,23 +26,30 @@ import (
...
+	defaultMetricPort                int32            = 8014
+	defaultPort                      int32            = 8013
+	defaultSocketPath                string           = ""
+	defaultEvaluator                 string           = "json"
+	defaultImage                     string           = "ghcr.io/open-feature/flagd"
+	defaultTag                       string           = "v0.3.1"
+	SyncProviderKubernetes           SyncProviderType = "kubernetes"
+	SyncProviderFilepath             SyncProviderType = "filepath"
+	SyncProviderHttp                 SyncProviderType = "http"
+	defaultSyncProvider                               = SyncProviderKubernetes
```
Comment (toddbaert) [inline apis/core/v1alpha1/flagsourceconfiguration_types.go:52]: Contants, we love to see 'em.

File: chart/open-feature-operator/README.md (hunk 2 above)
Comment (james-milligan) [reply chart/open-feature-operator/README.md]: Ive extended this comment to include a list

**Review by toddbaert — APPROVED** [review]

File: chart/open-feature-operator/README.md (hunk 2 above)
Comment (beeme1mr) [reply chart/open-feature-operator/README.md]: Did you push the changes?

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**